Add Arabic (Syria) localization support (ar-SY)

aboodgamer24 opened the pull request:
This PR adds complete Arabic language support to DLSS Swapper.

### Changes:
- Added Arabic (ar-SY) to supported languages in LanguageManager
- Created complete Arabic translation file (Resources.resw) with 998 translated strings
- Added Arabic language name display: "العربية"
- Included Arabic translations for all UI elements including:
  - Main application interface
  - Settings pages
  - Error messages and dialogs
  - Game library and DLL management features
  - Translation toolbox
  - Diagnostics and troubleshooting sections

### Testing:
- All Arabic translations have been verified for accuracy
- UI layout tested with Arabic text (RTL support)
- Language switching functionality confirmed working

This addition will make DLSS Swapper accessible to Arabic-speaking users worldwide.

aboodgamer24 commented:
Hi @beeradmoore , I hope you have a good day 
Yes you can delete it because i have added `LanguageManager.cs` into this PR.

aboodgamer24 commented:
<img width="1901" height="1055" alt="image" src="https://github.com/user-attachments/assets/72f1fa88-93ea-4802-8349-7a02cdd421fb" />
Ok!! I managed to make it RTL @beeradmoore

beeradmoore commented:
If we do make the change to do RTL out of the box I may not be able to ensure it's working correctly on every build. 

Would you say it is better for you as the end user to have RTL that may break/may not work for all designs, or just have the translation on the view that is left aligned? I am not sure what goes into keeping all designs to be working correctly when it is RTL.

aboodgamer24 commented:
I have added also list of Semitic language so if someone add Hebrew or Urdu it should work

aboodgamer24 commented:
<img width="1905" height="1062" alt="image" src="https://github.com/user-attachments/assets/ac617aac-06a0-4c75-a4ca-9da9a6e30377" />
<img width="1907" height="1068" alt="image" src="https://github.com/user-attachments/assets/ff1a129b-bcc1-46c4-99eb-28444d3c72ea" />
that's it another photos if want to see

aboodgamer24 commented:
Soo what do you think? should I merge it?

beeradmoore commented:
If it is working as intended feel free to add it in with the merge. I'll see what I can do about throwing it into a virtual machine or something and test with RTL to confirm things are working. 

Looks so very weird to me, but I have spent ~20 years seeing everything left aligned so it breaks my brain a little 😅

aboodgamer24 commented:
<img width="1909" height="1053" alt="image" src="https://github.com/user-attachments/assets/a0ced2ed-1e59-4a2e-a348-c4bf2337f205" />
Hi @beeradmoore , i realized that the photo cover is kind of flipped on RTL. Can you please give more time to fix it?

Thank you.

beeradmoore commented:
No worries. 

I hope it's as simple as setting flow direction on those images. There may be more places that needs to be fixed as there is multiple places to put that.

Also I was thinking there a few different windows that would need RTL support. I think its network testing window, diagnostics window, failed to launch window, and translation tools window

beeradmoore commented:
@aboodgamer24 , is says there will be some conflicts when merging with main. Don't worry about trying to fix those (if it shows on your end that is), I can fix before merging. It is just that I have been merging in translations today.

aboodgamer24 commented:
Hi @beeradmoore , 
<img width="2389" height="1060" alt="image" src="https://github.com/user-attachments/assets/a60d1875-f286-4c09-8fd0-cd8a29a8632b" />
 I have fixed the mirrored images and also added right-to-left formatting in the network test window. However, I couldn’t do the same for the translation tool box because I believe there is a major issue. For example, when I select Polish and go to the translation tool box, the language automatically switches to English. Other than that, everything seems to be working well on my end.

beeradmoore commented:
Thanks @aboodgamer24 , I will look over this soon and hopefully get it merged in soon

beeradmoore commented:
Thanks @aboodgamer24 , sorry for the delay. I tested this and it all seems to work as expected. I'll leave translation window for now. The main ones working is the main thing.

I appreciate the work you did to get RTL support. I was shocked how little it actually took code-wise.

aboodgamer24 commented:
> Thanks @aboodgamer24 , sorry for the delay. I tested this and it all seems to work as expected. I'll leave translation window for now. The main ones working is the main thing.
> 
> I appreciate the work you did to get RTL support. I was shocked how little it actually took code-wise.

You're welcome 😁
